charts/tidb-cluster/values.yaml

```diff
@@ -47,9 +47,20 @@ discovery:
 # if the ConfigMap was not changed.
 enableConfigMapRollout: true
 
+# Whether enable TLS connections between server nodes.
+# When enabled, PD/TiDB/TiKV will use TLS encrypted connections to transfer data between each node,
+# certificates will be generated automatically (if not already present).
+enableTLSServer: false
+
+# Whether enable TLS connection between TiDB server and MySQL client.
+# When enabled, TiDB will accept TLS encrypted connections from MySQL client, certificates will be generated
+# automatically.
+# Note: TLS connection is not forced on the server side, plain connections are also accepted after enableing.
+enableTLSClient: false
```

Comment:
`enableTLSClient` adds TLS support between tidb-server and MySQL clients and does not affect other components. Actually, it enables tidb-server server-side TLS support, `client` in its name is misleading. How about moving it to `tidb.enableTLS` field? What do you think? cc @weekface @tennix

Reply:
Yes, I agree. I prefer `tidb.enableTLS` because `tidb.tls` should be reserved if we want to add more tls related configurations into it.

Reply:
`tidb.enableTLS` is not clear, i prefer `tidb. enableTLSClient`

Reply:
but `tidb-server` does not act as a client. It enables server-side TLS support, then MySQL clients can connect to it. It's better not to introduce any new term when not necessary, no `client` term in tidb raw configuration too.

Reply:
I think the `client` word means the tidb server should validate the request's client cert.